### PR TITLE
UI Clean up

### PR DESCRIPTION
**Pull request by jamesstottmoj**
This PR makes the following changes:

- Adds a suite of unit tests
- Removes Govuk-Frontend-Django and upgrades Django to 5.1
- Uses a new base template rather than from the removed package
- Fixes missing favicon and apple touch moj logos
- Bumped several dependencies

### CODE REVIEW

File: templates/govuk-frontend.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+<head>
+  <meta charset="utf-8">
+  <title>
+    {% block title %}
+    {% endblock %}
+  </title>
+  <meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
+  {# Ensure that older IE versions always render with the correct rendering engine #}
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="theme-color" content="{{ themeColor | default:'#0b0c0c' }}" />
```
Comment (michaeljcollinsuk) [inline templates/govuk-frontend.html:14]: Should we get rid of `themeColour`? Presume its a hangover from control panel or is it used?

**Review by michaeljcollinsuk — CHANGES_REQUESTED** [review]
Tiny comment in the base template, otherwise looks good, thanks for the tests!

File: templates/govuk-frontend.html (hunk 1 above)
Comment (jamesstottmoj) [reply templates/govuk-frontend.html:14]: I think it's also used in the UI currently. Might be worth keeping it in?

Comment (michaeljcollinsuk) [reply templates/govuk-frontend.html:14]: Just cant tell see `themeColor` comes from, so will it always just use the default? Or does it come from the gov uk frontend?

Comment (jamesstottmoj) [reply templates/govuk-frontend.html:14]: By the looks of it, it's always set to the default. Theme colour is also in the govuk site. I don't think it hurts to keep it

Comment (michaeljcollinsuk) [reply templates/govuk-frontend.html:14]: Didnt mean remove the whole `<meta>` tag, just the use of reading from a context variable `themeColor` that isnt actually set anywhere. But no doesnt hurt to leave it I guess, just a bit confusing to see (and also it uses the wrong case)

**Review by michaeljcollinsuk — DISMISSED** [review]

**Review by michaeljcollinsuk — APPROVED** [review]

**Review by michaeljcollinsuk — APPROVED** [review]

**Pull request merged**